contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol

```diff
@@ -10,6 +10,31 @@ contract BasicAMB is BasicBridge {
     bytes constant internal SUBSIDIZED_MODE = bytes(abi.encodePacked("AMB-subsidized-mode"));
     bytes constant internal DEFRAYAL_MODE = bytes(abi.encodePacked("AMB-defrayal-mode"));
 
+    function initialize(
+        address _validatorContract,
+        uint256 _dailyLimit,
+        uint256 _maxPerTx,
+        uint256 _minPerTx,
+        uint256 _gasPrice,
+        uint256 _requiredBlockConfirmations
+    ) public returns(bool) {
+        require(!isInitialized());
+        require(_validatorContract != address(0));
+        require(_gasPrice > 0);
+        require(_requiredBlockConfirmations > 0);
+        require(_minPerTx >= 0 && _maxPerTx > _minPerTx && _dailyLimit > _maxPerTx);
+        addressStorage[keccak256(abi.encodePacked("validatorContract"))] = _validatorContract;
+        uintStorage[keccak256(abi.encodePacked("deployedAtBlock"))] = block.number;
+        uintStorage[keccak256(abi.encodePacked("dailyLimit"))] = _dailyLimit;
+        uintStorage[keccak256(abi.encodePacked("maxPerTx"))] = _maxPerTx;
+        uintStorage[keccak256(abi.encodePacked("minPerTx"))] = _minPerTx;
```

Comment:
we definitely don't need `_dailyLimit` abd `minPerTx` parameters
Probably we need to have something similar to `_maxPerTx` to limit gas consumption by one request.

Reply:
In this case `dailyLimit`, `minPerTx` and `maxPerTx` are used to limit the gas consumption. I reused the names of how they are stored to avoid defining new getter and setters.

https://github.com/poanetwork/poa-bridge-contracts/blob/ee1502762dcee3d2a6374103f99ed37aa93728f2/contracts/upgradeable_contracts/arbitrary_message/HomeAMB.sol#L10-L14

Probably `minPerTx` in this case doesn't apply so I should remove it. Regarding `dailyLimit` it was mentioned on the issue that it could be used, but if there is no use case maybe I should remove it too.

So on `requireToPassMessage` only a check for maximum gas will be performed. Sound good?

Reply:
Done